test/components/admin/currently_featured_tab_component_test.rb

```diff
@@ -47,98 +43,137 @@ class Admin::CurrentlyFeaturedTabComponentTest < ViewComponent::TestCase
+    assert_selector ".govuk-link", text: "Reorder documents", count: 0
   end
 
-  test "renders the correct row when the feature list item belongs to a offsite link" do
-    feature = build_stubbed(:feature, :with_offsite_link_association, feature_list: @feature_list)
-    title = feature.offsite_link.title
+  test "makes one call to the featurings FeaturedDocumentsTableComponent when featurings count is <= to maximum_featured_documents" do
+    topical_event = build_stubbed(:topical_event)
+    featurings = build_stubbed_list(:topical_event_featuring, @maximum_featured_documents, topical_event:)
+    table_component = Admin::TopicalEvents::Featurings::FeaturedDocumentsTableComponent.new(caption: "caption", featurings: [])
+
+    Admin::TopicalEvents::Featurings::FeaturedDocumentsTableComponent
+    .expects(:new)
+    .with(featurings:, caption: "#{featurings.count} featured documents live on GOV.UK")
+    .once
+    .returns(table_component)
+
+    table_component
+    .expects(:render)
+    .returns("")
 
     render_inline(Admin::CurrentlyFeaturedTabComponent.new(
-                    features: [feature],
+                    featurings:,
                     maximum_featured_documents: @maximum_featured_documents,
                   ))
-
-    assert_equal page.all(".govuk-table .govuk-table__row .govuk-table__cell")[0].text, title
-    assert_equal page.all(".govuk-table .govuk-table__row .govuk-table__cell")[1].text, "Alert (offsite link)"
-    assert_equal page.all(".govuk-table .govuk-table__row .govuk-table__cell")[2].text, ""
-
-    actions_column = page.all(".govuk-table .govuk-table__row .govuk-table__cell")[3]
-    actions_column.assert_selector "a[href='#{polymorphic_path([:edit, :admin, feature.offsite_link.parent, feature.offsite_link])}']", text: "Edit #{title}"
-    actions_column.assert_selector "a[href='#{confirm_unfeature_admin_feature_list_feature_path(feature.feature_list, feature)}']", text: "Unfeature #{title}"
   end
 
-  test "renders the correct row when the feature list item does not have an association" do
-    feature = build_stubbed(:feature, document: nil)
+  test "makes two calls to the featurings FeaturedDocumentsTableComponent when featurings count is greater than maximum_featured_documents" do
+    topical_event = build_stubbed(:topical_event)
+    live_featurings = build_stubbed_list(:topical_event_featuring, @maximum_featured_documents, topical_event:)
+    remaining_featurings = build_stubbed(:topical_event_featuring, topical_event:)
+    table_component1 = Admin::TopicalEvents::Featurings::FeaturedDocumentsTableComponent.new(caption: "caption1", featurings: [])
+    table_component2 = Admin::TopicalEvents::Featurings::FeaturedDocumentsTableComponent.new(caption: "caption2", featurings: [])
+
+    Admin::TopicalEvents::Featurings::FeaturedDocumentsTableComponent
```

Comment:
did wonder if there is a way to chain these two stubbed expectations on the FeaturedDocumentsTableComponent, rather than setting up again, to tidy up the code?